tensorflow/python/data/experimental/kernel_tests/data_service_ops_test.py

```diff
@@ -680,6 +713,24 @@ def key_func(x):
         self.assertAllEqual(next(it).numpy(), element)
     self.assertEmpty(list(it))
 
+  # @combinations.generate(test_base.eager_only_combinations())
+  # def testCyclicDistribute(self):
+
+  #   cluster_1 = self.create_cluster(num_workers=1)
+  #   cluster_2 = self.create_cluster(num_workers=1)
+  #   num_sizes = 10
+  #   size_repeats = 5
+  #   numbers = [1 * i for i in range(num_sizes)] * size_repeats
+  #   ds = dataset_ops.Dataset.from_tensors(numbers)
+  #   ds = self.make_distributed_dataset(ds, cluster_1, processing_mode="parallel_epochs")
+  #   ds = ds.map(lambda x: x + 1)
+  #   ds = self.make_distributed_dataset(ds, cluster_2, processing_mode="parallel_epochs")
+  #   ds = ds.map(lambda x: x - 1)
+  #   ds = self.make_distributed_dataset(
+  #       ds, cluster_1, processing_mode="parallel_epochs", job_name="temp_jobname")
+
+  #   self.assertDatasetProduces(ds, numbers, assert_items_equal=True)
```

Comment:
@aaudiber I was trying to experiment on a cyclic distribute using this test and encountered a timeout instead of a failure. Any idea why this is the case? 
P.S: This was just out of curiosity.

Reply:
This looks like a deadlock triggered when a worker's GetNext request requires calling back to the same worker. Worker's hold a lock during GetNext: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/data/service/worker_impl.cc#L184

I have an unrelated outstanding CL that will relax the locking so that the lock isn't held while calling Iterator::GetNext. That should fix this deadlock

Reply:
Thanks for the info. Will add that test once the deadlock is handled. Also, I have removed this commented test for final review.